### DISCUSSION TITLE
Zigbee2Tasmota how to request sensor last state

### DISCUSSION THREAD

**Discussion by nagyrobi** [arendst/Tasmota#11008] [Q&A] [answered]
Many Zigbee sensors publish their values only on significant change, and very large periods of time can pass without any value update from the sensor itself. This is specially the case with [BlitzWolf BW-IS4](https://zigbee.blakadder.com/BlitzWolf_BW-IS4.html) temp/hum and its clones. The problem is that for example after a Home Assistant restart or after any downtime of some components between Tasmota and the Home Automation system the values can be lost and there's no way to retrieve them. 

But as I've seen Tasmota retains these values very well. It shows the last temp/hum values using ZbStatus3 command and of course they can be seen in the web interface too. These survive Tasmota reboot, power outage etc.
There is the good old [SetOption59](https://tasmota.github.io/docs/Commands/#setoption59) which allows for a query of the states on demand but that only covers switches and lights. And there's the issue that the value is published at a different MQTT topic than the sensor and it's hardly possible (in Home Assistant at least) to set the value of a sensor from multiple topics and payloads.

I am running Zigbee2Tasmota on a Sonoff ZbBridge, using Tasmota 9.2.0.7 self-compiled. 

It would be very useful to have a command to trigger new `tele/zigbee/SENSOR` publications to get the actual values of all the sensors cached in Tasmota on demand. We could use that for example at Home Assistant boot. So this way all the states would be synced immediately with the last known values. There is some basic information [in the docs](https://tasmota.github.io/docs/Home-Assistant/#useful-automations) but `zbread` doesn't seem to be supported anymore:
```
17:50:39.415 CMD: zbread
17:50:39.420 MQT: stat/zigbee1/RESULT = {"Command":"Unknown"}
```
Seen various issue topics that there should also be a `zigbeeread` command, that doesn't exist either:
```
17:50:36.665 CMD: zigbeeread
17:50:36.670 MQT: stat/zigbee1/RESULT = {"Command":"Unknown"}
```
Is there a way to sync the last state of the sensors on demand? Hopefully on the same  `tele/zigbee/SENSOR` topic where normal pushes come in?

**Comment by s-hadinger** [comment, marked as answer]
Tasmota only caches a selected number of sensor value because of memory constraints of ESP8266. You can get all values with `ZbInfo <device>`

Indeed `ZbRead` was replaced with `ZbSend {"Device":"<device","Read":{...}}`. This will send a read command to the device that may or may not return a value. Battery powered devices generally don't respond, as well as Zigbee Tuya since the Tuya protocol has no 'read' command.

**Reply by nagyrobi** [reply]
Can you please elaborate a little bit more on the read command exact syntax? I'm having a hard time understanding the [manual](https://tasmota.github.io/docs/Zigbee/#sending-device-commands). My device is:
`20:39:11.145 MQT: stat/zigbee1/RESULT = {"ZbStatus3":[{"Device":"0x888C","Name":"Room1","IEEEAddr":"0x847127FFFE0CFEE6","ModelId":"TS0201","Manufacturer":"_TZ2000_a476raq2","Endpoints":[1],"Config":["T01"],"Temperature":23.6,"Humidity":42,"Reachable":true,"BatteryPercentage":100,"LastSeen":283,"LastSeenEpoch":1613590468,"LinkQuality":34}]}`

Also is there a specific limit on the cached sensor numbers? How should I plan, how many sensors could I use?

**Reply by nagyrobi** [reply]
You mean like this?
`zbsend {"Device":"0x888C","Read":{"Temperature":true}}`
The answer is:
`21:55:35.044 MQT: tele/zigbee1/RESULT = {"ZbRouteError":{"ShortAddr":"0x888C","Status":66,"StatusMessage":"MAC_INDIRECT_TIMEOUT"}}`
